src/weight_windows.cpp

```diff
@@ -8,251 +8,276 @@
-  xml_document doc;
+
+  std::string filename = settings::path_input + "variance_reduction.xml";
+
+  // Parse variance_reduction.xml file
+  pugi::xml_document doc;
 
   // try the read vr file
   if (file_exists(filename)) {
     auto result = doc.load_file(filename.c_str());
-    
+
     if (!result) {
-      fatal_error("Error processing variance_reduction.xml file.");             
+      fatal_error("Error processing variance_reduction.xml file.");
     }
-    // Get root element                                                                     
-    xml_node root = doc.document_element();
+    // Get root element
+    pugi::xml_node root = doc.document_element();
 
     // Display output message
     write_message("Reading Variance Reduction XML file...", 5);
 
     // check for variance_reduction
-    if (check_for_node(root, "variance_reduction")) {
-      // check for weight window section
-      xml_node variance_reduction = root.child("variance_reduction");
-   
-      if (check_for_node(variance_reduction,"weight_windows"))
-	xml_node weight_windows = variance_reduction.child("weight_windows");
-      else
-	// Display output message
-	warning("variance_reduction file has been read, but no variance reduction has been enabled.");
-	
-    else
-      fatal_error("variance_reduction element is missing from the variance_reduction.xml file");             
-
-    // try and read a weight window
-    read_weight_windows(weight_windows);
-  }                                                                        
+    if (!check_for_node(root, "variance_reduction")) {
+      fatal_error("variance_reduction element is missing from the "
+                  "variance_reduction.xml file");
+    }
+    pugi::xml_node vr_node = root.child("variance_reduction");
+
+    // check for weight window section
+    if (check_for_node(vr_node, "weight_windows")) {
+      pugi::xml_node weight_windows = vr_node.child("weight_windows");
+      read_weight_windows(weight_windows);
+    } else {
+      // Display output message
+      warning("variance_reduction file has been read, but no variance "
+              "reduction has been enabled.");
+    }
+  }
 }
 
 // read the weight window section of a vr file
-void read_weight_windows(pugi::xml_node node) 
+void read_weight_windows(pugi::xml_node node)
 {
   using namespace pugi;
 
   // read all the meshes in the VR file
   read_meshes(node);
 
   // make sure we have at least one settings section
-  if (check_for_node(node, "settings")) {
-    for (auto settings : node.children("settings")) {
-      // do i need to check for id?
-      int id = get_node_value(node, "id");
-
-      std::shared_ptr<WeightWindowParameters> paramters = read_settings(settings);
-      ww_map[id] = ww_params.size() - 1; // assign the id of the settings object
-      ww_params.emplace_back(ww_params); // put the window in the map
+  if (check_for_node(node, "weight_window_settings")) {
```

Comment:
did you change the name of from settings -> weight_window_settings in the variance_reduction.py module?

Reply:
Yeah I did. It was kind of an arbitrary change. I was thinking it might make sense to have those nodes go directly under the `<variance_reduction>` node in the XML file to keep the number of levels lower. What do you think?

Reply:
so there is now no <weight_window> section?

Reply:
Not at the moment, no. I realized that the only thing that would end up in there are the settings and domain nodes (meshes should probably appear at the root level under `<variance_reduction>`, but maybe it does make sense to keep the `<weight_window>` section. I can change it back.

Reply:
Went ahead and reverted the names on the C++ side. They are now `domain` and `settings` again. Added a `<weight_window>` section to the `variance_reduction.xml` file on the Python side.